examples/buffer/src/main.rs

```diff
@@ -18,13 +16,13 @@ fn main() -> Result<()> {
     assert_eq!(reference.Capacity()?, 11);
 
     // Write to buffer...
-    {
+    unsafe {
         let slice = as_slice(&reference)?;
         slice.copy_from_slice(b"hello world");
```

Comment:
As with the D3D12 example, should we aim for as little as possible code in an unsafe block?

```suggestion
    {
        let slice = unsafe { as_slice(&reference) }?;
        slice.copy_from_slice(b"hello world");
```

Same below if you think this is worth it.